staging/src/k8s.io/api/apps/v1/types.go

```diff
@@ -24,9 +24,56 @@ import (
 
 const (
 	ControllerRevisionHashLabelKey = "controller-revision-hash"
+	StatefulSetRevisionLabel       = ControllerRevisionHashLabelKey
 	DeprecatedTemplateGeneration   = "deprecated.daemonset.template.generation"
 )
 
+// ScaleSpec describes the attributes of a scale subresource
+type ScaleSpec struct {
```

Comment:
@DirectXMan12 @deads2k for propagation of scale into apps/v1... I thought we really didn't want to do this

Reply:
#49504 indicates that this is still not clear. Currently we have an individual scale in each group version. If  there is a plan to change this in 1.9 and migrate to a different strategy it is not concrete afaik

Reply:
the current approach has known problems (HPA must hardcode custom clients for everything it is asked to scale) and scale was added to the apps group without considering how it would actually interact with autoscaling. we should complete the design that accounts for that *before* propagating those problems into a stable API group.

Reply:
Agreed that this warrants more discussion. Added a TODO on this PR.

Reply:
We did consider the auto-scaling interaction when adding Scale to the apps group, as well as the interaction with eviction controller and pod disruption budgets. In  #42752 we decided to add a scale sub resource for all object that made sense. If we are going to pivot on the implementation, that is fine, and importing the scale subresource from another location will be relatively minor change. If the duck typing approach form #49504  is taken, I still think we should support the scale subresource in the apps group because we have it in extensions/v1beta1 and users of Deployment may have built expectations about it into their code.

Reply:
> We did consider the auto-scaling interaction when adding Scale to the apps group

An approach that required hard-coding HPA to yet another API group's Scale kind seems like a mistake.

> In #42752 we decided to add a scale sub resource for all object that made sense.

A scale subresource makes sense. Representing it with a different group/version/kind that must still have the same schema does not.

> If the duck typing approach form #49504 is taken...

I strongly hope it is not. Nothing else in our API uses duck-typing, I don't think we should start promoting it now.

>  I still think we should support the scale subresource in the apps group because we have it in extensions/v1beta1

The scale subresource is a good idea, but I have yet to see a compelling reason to define *different* Scale kinds in multiple API groups that are required to have identical schemas for duck typing. That seems like the worst of both worlds... you can't have a generic scale client that can use the same Scale kind against all `/scale` subresources, yet you don't gain any flexibility in having per-group Scale kinds because of duck typing constraints.

Before merging yet another Scale type, is there a reason not to settle on a common Scale kind?

Reply:
> An approach that required hard-coding HPA to yet another API group's Scale kind seems like a mistake.

Perhaps, but if we decide to replace it with a kind from a different group version rather than using one resource per group the PR to implement, imo, can be handled with a follow up.

> Before merging yet another Scale type (in a long-term supported version, no less), is there a reason not to settle on a common Scale kind?

We want to issue and merge the PRs to promote the workloads API. There will be  follow up PRs to add Deployment, to add ReplicaSet, to add Condition types #51594, and to deal with any other outstanding issues. If we choose a different, and definitive, home for the scale sub-resource I would prefer to deal with it in a follow up PR (which should be trivial) rather than blocking all other development pending resolution of the conversation in #49504.

Reply:
> If we choose a different, and definitive, home for the scale sub-resource I would prefer to deal with it in a follow up PR

We shouldn't merge apps/v1 Scale until this question is settled. It can be pulled out of this PR to unblock it.